.github/workflows/publish-indy.yml

```diff
@@ -0,0 +1,99 @@
+name: Publish ACA-Py (Indy)
+on:
+  release:
+    types: [released]
+  workflow_dispatch:
+    inputs:
+      indy_version:
+        description: 'Indy SDK Version'
+        required: false
+        type: string
+      tag:
+        description: 'Image tag'
+        required: true
+        type: string
+      version:
+        description: "Version label in image"
+        required: true
+        type: string
+
+env:
+  INDY_VERSION: 1.16.0
+
+jobs:
+  publish-image:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+
+    name: Publish (Indy)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Gather image info
+        id: info
+        run: |
+          echo "::set-output name=repo-owner::${GITHUB_REPOSITORY_OWNER,,}"
+
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to the GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Image Metadata (manual)
+        if: github.event_name == 'workflow_dispatch'
+        id: dispatch-meta
+        uses: docker/metadata-action@v3
+        with:
+          images: |
+            ghcr.io/${{ steps.info.outputs.repo-owner }}/aries-cloudagent-python
+          tags: |
+            type=raw,value=${{ inputs.tag }}
+
+      - name: Setup Image Metadata (release)
+        if: github.event_name == 'release'
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: |
+            ghcr.io/${{ steps.info.outputs.repo-owner }}/aries-cloudagent-python
+          tags: |
+            type=semver,pattern=py${{ matrix.python-version }}-indy-{{ inputs.indy_version || env.INDY_VERSION }}-{{version}}
+
+      - name: Build and Push Image to ghcr.io
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          context: .
+          file: docker/Dockerfile.indy
+          tags: ${{ steps.dispatch-meta.outputs.tags || steps.meta.outputs.tags }}
+          target: main
+          build-args: |
+            python_version=${{ matrix.python-version }}
+            indy_version=${{ inputs.indy_version || env.INDY_VERSION }}
+            acapy_version=${{ inputs.version || github.event.release.tag_name }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
```

Comment:
Setup of the image tags is not working as expected.  When run manually or triggered by a release a single tag is used for each of the images, resulting in the images being overwritten with only the last matrix job to run as the final image tagged.

Whether run manually or triggered by a release there should be an explicitly tagged image for each version of python listing in the matrix. 

References:
https://github.com/WadeBarnes/aries-cloudagent-python/actions/runs/3517366030
https://github.com/WadeBarnes/aries-cloudagent-python/actions/runs/3517511271
https://github.com/WadeBarnes/aries-cloudagent-python/pkgs/container/aries-cloudagent-python

Reply:
I believe this has been corrected now. See: https://github.com/dbluhm/aries-cloudagent-python/actions/runs/3603865313/jobs/6072572552